### DISCUSSION TITLE
How to reply to a top-level comment?

### DISCUSSION THREAD

**Discussion by ktfleming** [wandersoncferreira/code-review#78] [Q&A] [answered]
I'm having some trouble replying to a top-level comment on a PR (like [this one](https://github.com/wandersoncferreira/code-review/pull/50#issuecomment-976183170)). It seems like no matter where point is (on the line labeled "Conversation", on the line that includes the Github user name and the timestamp, or on the comment itself), calling `code-review-comment-add-or-edit` just gives the error `You can't add text over unspecified region`. I think that's the correct command that should let me reply to a comment, right? Maybe there's something I'm missing or doing wrong...I'm not sure if this is a bug or just my mistake, so I'm trying a discussion instead of an issue, at least to start.

**Comment by ktfleming** [comment, marked as answer]
Ah wait, right after I posted this I just realized that maybe it's not really a "reply" that I want to do as it seems that top-level comments can't have direct replies. What I wanted to do was the equivalent of scrolling to the bottom of the main PR page, typing a comment into the text area, and hitting the "Comment" button. Does code-review support this? I suppose I could re-review the PR and add a comment at the same time, but really I just would like to add a comment by itself.

**Reply by wandersoncferreira** [reply]
As usual, you are right!

This should be easy to include now, there are two possible behaviors for this feature:

1. Add as many comments as you like in the buffer and then call "send comments"
2. Send the comment as you add then to the buffer

I like the first approach better, but I can see the advantage of adding a single comment and on `C-c C-c` I already updated upstream. wdyt?

**Reply by ktfleming** [reply]
I think the first approach sounds good! When you say you updated upstream, do you mean that you already added this feature in? I pulled the latest version and looked around, but I couldn't see a way to add a single comment yet...just as an example, I wrote a test comment (from Github) that shows what I'm trying to achieve: https://github.com/ktfleming/code-review-bug/pull/1#issuecomment-983110988

**Reply by ktfleming** [reply]
Oh wait! Sorry, I think this whole thing was my misunderstanding...it seems like I can get what I want by simply adding feedback and then choosing "comment" from the transient menu.

**Reply by wandersoncferreira** [reply]
> Oh wait! Sorry, I think this whole thing was my misunderstanding...it seems like I can get what I want by simply adding feedback and then choosing "comment" from the transient menu.

I thought this too... but there is a difference in meaning to Github on these comments. If you add a feedback and send a Comment review it will be assigned by GitHub as a "PullRequestReview" type and if you only add a single comment it will be assigned as "IssueComment".

This distinction is used here https://github.com/wandersoncferreira/code-review/blob/a1e01be01bd102ea3b9496b928199efc6615de2c/code-review-github.el#L629 to create correct links if you desire to promote a comment to a new issue.

**Reply by wandersoncferreira** [reply]
if you are interested in developing this feature I can provide some guidance around the codebase... it is not hard/difficult but unfortunately is not an isolated change as I would like it to be.

let me know.

**Reply by ktfleming** [reply]
I see! Thanks for the explanation. I think for now the way of commenting that I described above should work for my purposes!

I suppose I'll mark this thread as an answer then.
